### PR TITLE
[Merged by Bors] - chore(Algebra/Prod): add `Prod.mk_one_one`

### PR DESCRIPTION
**Pull request by urkud**
- Normalize `(1, 1) ` and `(0, 0)` to `1` and `0`, respectively.
- Add `Polynomial.mul_coeff_one`

Moves:
- PerfectClosure.mk_zero_zero -> PefectClosure.mk_zero
- PerfectClosure.mk_zero -> PerfectClosure.mk_zero_right

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Counterexamples/CanonicallyOrderedCommSemiringTwoMul.lean [hunk 1]
```diff
@@ -134,12 +134,14 @@ theorem mul_L {a b : ℕ × ZMod 2} (ha : a ≠ (0, 1)) (hb : b ≠ (0, 1)) : a
   rcases a with ⟨a, a2⟩
   rcases b with ⟨b, b2⟩
   cases b
-  · rcases mem_zmod_2 b2 with (rfl | rfl) <;> rcases mem_zmod_2 a2 with (rfl | rfl) <;> simp
+  · rcases mem_zmod_2 b2 with (rfl | rfl) <;> rcases mem_zmod_2 a2 with (rfl | rfl) <;>
+      simp [-Prod.mk_zero_zero]
```
Comment (mattrobball) [inline Counterexamples/CanonicallyOrderedCommSemiringTwoMul.lean:138]:
```suggestion
      simp only [Prod.mk_mul_mk, mul_zero, mul_one, ne_eq, Prod.mk.injEq, zero_ne_one, and_false, 
        not_false_eq_true, not_true_eq_false]
```

File: Counterexamples/CanonicallyOrderedCommSemiringTwoMul.lean [hunk 2]
```diff
@@ -134,12 +134,14 @@ theorem mul_L {a b : ℕ × ZMod 2} (ha : a ≠ (0, 1)) (hb : b ≠ (0, 1)) : a
...
     -- while this looks like a non-terminal `simp`, it (almost) isn't: there is only one goal where
     -- it does not finish the proof and on that goal it asks to prove `false`
     exact hb rfl
   cases a
-  · rcases mem_zmod_2 b2 with (rfl | rfl) <;> rcases mem_zmod_2 a2 with (rfl | rfl) <;> simp
+  · rcases mem_zmod_2 b2 with (rfl | rfl) <;> rcases mem_zmod_2 a2 with (rfl | rfl) <;>
+      simp [-Prod.mk_zero_zero]
```
Comment (mattrobball) [inline Counterexamples/CanonicallyOrderedCommSemiringTwoMul.lean:144]:
```suggestion
            simp only [Prod.mk_mul_mk, mul_zero, zero_mul, mul_one, ne_eq, Prod.mk.injEq, zero_ne_one, and_false, 
        not_false_eq_true, not_true_eq_false]
```

File: Counterexamples/CanonicallyOrderedCommSemiringTwoMul.lean (hunk 1 above)
Comment (mattrobball) [reply Counterexamples/CanonicallyOrderedCommSemiringTwoMul.lean:138]: We should delete the comment that tries to talk you into violating your principles.

File: Counterexamples/CanonicallyOrderedCommSemiringTwoMul.lean [hunk 3]
```diff
@@ -134,12 +134,14 @@ theorem mul_L {a b : ℕ × ZMod 2} (ha : a ≠ (0, 1)) (hb : b ≠ (0, 1)) : a
   rcases a with ⟨a, a2⟩
   rcases b with ⟨b, b2⟩
   cases b
-  · rcases mem_zmod_2 b2 with (rfl | rfl) <;> rcases mem_zmod_2 a2 with (rfl | rfl) <;> simp
+  · rcases mem_zmod_2 b2 with (rfl | rfl) <;> rcases mem_zmod_2 a2 with (rfl | rfl) <;>
+      simp [-Prod.mk_zero_zero]
     -- while this looks like a non-terminal `simp`, it (almost) isn't: there is only one goal where
     -- it does not finish the proof and on that goal it asks to prove `false`
```
Comment (urkud) [inline Counterexamples/CanonicallyOrderedCommSemiringTwoMul.lean:140]:
```suggestion
```

File: Counterexamples/CanonicallyOrderedCommSemiringTwoMul.lean [hunk 4]
```diff
@@ -134,12 +134,14 @@ theorem mul_L {a b : ℕ × ZMod 2} (ha : a ≠ (0, 1)) (hb : b ≠ (0, 1)) : a
...
     exact hb rfl
   cases a
-  · rcases mem_zmod_2 b2 with (rfl | rfl) <;> rcases mem_zmod_2 a2 with (rfl | rfl) <;> simp
+  · rcases mem_zmod_2 b2 with (rfl | rfl) <;> rcases mem_zmod_2 a2 with (rfl | rfl) <;>
+      simp [-Prod.mk_zero_zero]
     -- while this looks like a non-terminal `simp`, it (almost) isn't: there is only one goal where
     -- it does not finish the proof and on that goal it asks to prove `false`
```
Comment (urkud) [inline Counterexamples/CanonicallyOrderedCommSemiringTwoMul.lean:146]:
```suggestion
```

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**